--- qtbase/src/3rdparty/pcre2/CMakeLists.txt	2026-03-05 17:51:49.000000000 +1000
+++ qtbase/src/3rdparty/pcre2/CMakeLists.txt	2026-04-04 16:30:55.996748110 +1000
@@ -45,6 +45,7 @@
         src/pcre2_xclass.c
     DEFINES
         HAVE_CONFIG_H
+        PCRE2_DISABLE_JIT
     PUBLIC_DEFINES
         PCRE2_CODE_UNIT_WIDTH=16
     PUBLIC_INCLUDE_DIRECTORIES
@@ -56,23 +57,8 @@
 ## Scopes:
 #####################################################################
 
-qt_internal_extend_target(BundledPcre2 CONDITION QNX OR UIKIT
-    DEFINES
-        PCRE2_DISABLE_JIT
-)
-
-qt_internal_extend_target(BundledPcre2 CONDITION (TEST_architecture_arch STREQUAL "arm") AND WIN32
-    DEFINES
-        PCRE2_DISABLE_JIT
-)
-
-qt_internal_extend_target(BundledPcre2 CONDITION (TEST_architecture_arch STREQUAL "arm64") AND WIN32
-    DEFINES
-        PCRE2_DISABLE_JIT
-)
-
 if (APPLE)
-    target_compile_options(BundledPcre2 PRIVATE "SHELL:-Xarch_arm64 -DPCRE2_DISABLE_JIT")
+    target_compile_options(BundledPcre2 PRIVATE "SHELL:-Xarch_arm64")
 endif()
 
 qt_internal_extend_target(BundledPcre2 CONDITION WIN32
