--- a/src/linguist/CMakeLists.txt
+++ b/src/linguist/CMakeLists.txt
@@ -22,7 +22,7 @@ add_subdirectory(lupdate-pro)
 add_subdirectory(ltext2id)
 if(QT_FEATURE_process AND QT_FEATURE_pushbutton AND QT_FEATURE_toolbutton
     AND QT_FEATURE_png AND QT_FEATURE_mdiarea AND QT_FEATURE_syntaxhighlighter
-    AND TARGET Qt::Widgets AND TARGET Qt::Quick)
+    AND TARGET Qt::Network AND TARGET Qt::Widgets)
     add_subdirectory(linguist)
 endif()
 
--- a/src/linguist/linguist/CMakeLists.txt
+++ b/src/linguist/linguist/CMakeLists.txt
@@ -44,7 +44,6 @@ qt_internal_add_app(linguist
         errorsview.cpp errorsview.h
         finddialog.cpp finddialog.h finddialog.ui
         uiformpreviewview.cpp uiformpreviewview.h
-        qmlformpreviewview.h qmlformpreviewview.cpp
         globals.cpp globals.h
         main.cpp
         mainwindow.cpp mainwindow.h mainwindow.ui
@@ -77,10 +76,9 @@ qt_internal_add_app(linguist
         Qt::CorePrivate
         Qt::Gui
         Qt::GuiPrivate
+        Qt::Network
         Qt::UiToolsPrivate
         Qt::Widgets
-        Qt::QuickWidgets
-        Qt::QuickLayouts
 )
 
 qt_internal_extend_target(linguist CONDITION QT_FEATURE_process
--- a/src/linguist/linguist/mainwindow.cpp
+++ b/src/linguist/linguist/mainwindow.cpp
@@ -14,7 +14,6 @@
 #include "finddialog.h"
 #include <helpclient.h>
 #include "uiformpreviewview.h"
-#include "qmlformpreviewview.h"
 #include "globals.h"
 #include "messageeditor.h"
 #include "messagemodel.h"
@@ -72,11 +71,6 @@ static bool hasUiFormPreview(const QString &fileName)
     return fileName.endsWith(".ui"_L1) || fileName.endsWith(".jui"_L1);
 }
 
-static bool hasQmlFormPreview(const QString &fileName, bool qmlPreviewChecked)
-{
-    return fileName.endsWith(QLatin1String(".qml")) && qmlPreviewChecked;
-}
-
 static const int MessageMS = 2500;
 
 } // namespace
@@ -352,11 +346,9 @@ MainWindow::MainWindow(HelpClientType helpClientType)
     m_sourceAndFormView = new QStackedWidget(this);
     m_sourceAndFormDock->setWidget(m_sourceAndFormView);
     m_uiFormPreviewView = new UiFormPreviewView(0, m_dataModel);
-    m_qmlFormPreviewView = new QmlFormPreviewView(m_dataModel);
     m_sourceCodeView = new SourceCodeView(0);
     m_sourceAndFormView->addWidget(m_sourceCodeView);
     m_sourceAndFormView->addWidget(m_uiFormPreviewView);
-    m_sourceAndFormView->addWidget(m_qmlFormPreviewView);
 
     // Set up errors dock widget
     m_errorsDock = new QDockWidget(this);
@@ -565,7 +557,6 @@ void MainWindow::modelCountChanged()
     m_ui.actionFindPrev->setEnabled(false);
 
     m_uiFormPreviewView->setSourceContext(-1, 0);
-    m_qmlFormPreviewView->setSourceContext(-1, 0);
     updateVisibleColumns();
 }
 
@@ -1606,9 +1597,6 @@ void MainWindow::translationChanged(const MultiDataIndex &index)
     MessageItem *m = m_dataModel->messageItem(index);
     if (hasUiFormPreview(m->fileName()))
         m_uiFormPreviewView->setSourceContext(index.model(), m);
-    else if (hasQmlFormPreview(m->fileName(), m_ui.actionQmlPreview->isChecked()))
-        if (!m_qmlFormPreviewView->setSourceContext(index.model(), m))
-            m_ui.actionQmlPreview->setChecked(false);
 }
 
 // This and the following function operate directly on the messageitem,
@@ -1624,10 +1612,6 @@ void MainWindow::updateTranslation(const QStringList &translations)
     m->setTranslations(translations);
     if (!m->fileName().isEmpty() && hasUiFormPreview(m->fileName()))
         m_uiFormPreviewView->setSourceContext(m_currentIndex.model(), m);
-    else if (!m->fileName().isEmpty()
-             && hasQmlFormPreview(m->fileName(), m_ui.actionQmlPreview->isChecked()))
-        if (!m_qmlFormPreviewView->setSourceContext(m_currentIndex.model(), m))
-            m_ui.actionQmlPreview->setChecked(false);
     updateDanger(m_currentIndex, true);
 
     if (m->isFinished())
@@ -2056,7 +2040,6 @@ void MainWindow::setupMenuBar()
     connect(m_ui.actionDisplayGuesses, &QAction::triggered,
             m_phraseView, &PhraseView::toggleGuessing);
     connect(m_ui.actionStatistics, &QAction::triggered, this, &MainWindow::showStatistics);
-    connect(m_ui.actionQmlPreview, &QAction::triggered, this, &MainWindow::toggleQmlPreview);
     connect(m_ui.actionVisualizeWhitespace, &QAction::triggered,
             this, &MainWindow::toggleVisualizeWhitespace);
     connect(m_ui.actionIncreaseZoom, &QAction::triggered,
@@ -2166,11 +2149,7 @@ void MainWindow::updateSourceView(int model, MessageItem *item)
         if (hasUiFormPreview(item->fileName())) {
             m_sourceAndFormView->setCurrentWidget(m_uiFormPreviewView);
             m_uiFormPreviewView->setSourceContext(model, item);
-        } else if (hasQmlFormPreview(item->fileName(), m_ui.actionQmlPreview->isChecked())
-                   && m_qmlFormPreviewView->setSourceContext(model, item)) {
-            m_sourceAndFormView->setCurrentWidget(m_qmlFormPreviewView);
         } else {
-            m_ui.actionQmlPreview->setChecked(false);
             m_sourceAndFormView->setCurrentWidget(m_sourceCodeView);
             QDir dir = QFileInfo(m_dataModel->srcFileName(model)).dir();
             QString fileName = QDir::cleanPath(dir.absoluteFilePath(item->fileName()));
@@ -2774,14 +2753,6 @@ void MainWindow::showStatistics()
     updateStatistics();
 }
 
-void MainWindow::toggleQmlPreview()
-{
-    if (m_ui.actionQmlPreview->isChecked())
-        m_sourceAndFormView->setCurrentWidget(m_qmlFormPreviewView);
-    else
-        m_sourceAndFormView->setCurrentWidget(m_sourceCodeView);
-}
-
 void MainWindow::toggleVisualizeWhitespace()
 {
     m_messageEditor->setVisualizeWhitespace(m_ui.actionVisualizeWhitespace->isChecked());
--- a/src/linguist/linguist/mainwindow.h
+++ b/src/linguist/linguist/mainwindow.h
@@ -37,7 +37,6 @@ class ErrorsView;
 class FocusWatcher;
 class HelpClient;
 class UiFormPreviewView;
-class QmlFormPreviewView;
 class MessageEditor;
 class PhraseView;
 class SourceCodeView;
@@ -139,7 +138,6 @@ private slots:
                   FindDialog::FindOptions options, int statusFilter);
     void revalidate();
     void showStatistics();
-    void toggleQmlPreview();
     void toggleVisualizeWhitespace();
     void onWhatsThis();
     void updatePhraseDicts();
@@ -215,7 +213,6 @@ private:
     QTabWidget *m_contextAndLabelView;
     SourceCodeView *m_sourceCodeView;
     UiFormPreviewView *m_uiFormPreviewView;
-    QmlFormPreviewView *m_qmlFormPreviewView;
     ErrorsView *m_errorsView;
     QLabel *m_progressLabel;
     QLabel *m_modifiedLabel;
--- a/src/linguist/linguist/mainwindow.ui
+++ b/src/linguist/linguist/mainwindow.ui
@@ -98,7 +98,6 @@
     </widget>
     <addaction name="actionResetSorting"/>
     <addaction name="actionDisplayGuesses"/>
-    <addaction name="actionQmlPreview"/>
     <addaction name="actionLengthVariants"/>
     <addaction name="actionVisualizeWhitespace"/>
     <addaction name="separator"/>
@@ -649,26 +648,6 @@
     <enum>QAction::MenuRole::NoRole</enum>
    </property>
   </action>
-  <action name="actionQmlPreview">
-   <property name="checkable">
-    <bool>true</bool>
-   </property>
-   <property name="checked">
-    <bool>true</bool>
-   </property>
-   <property name="text">
-    <string>&amp;QML preview</string>
-   </property>
-   <property name="toolTip">
-    <string>Displays a preview of QML documents.</string>
-   </property>
-   <property name="whatsThis">
-    <string/>
-   </property>
-   <property name="menuRole">
-    <enum>QAction::MenuRole::NoRole</enum>
-   </property>
-  </action>
   <action name="actionManual">
    <property name="text">
     <string>&amp;Manual</string>
